### PR TITLE
ci: create deployment script to deploy + verify base contracts on mainnet

### PR DESCRIPTION
**Pull request by CJ42**
# What does this PR introduce?

## 🤖 CI

Create CI script that deploy + verify base contracts on LUKSO mainnet.

These contracts are:

- `LSP1UniversalReceiverDelegateUP`
- `UniversalProfileInit`
- `LSP6KeyManagerInit`
- `LSP7MintableInit`
- `LSP8MintableInit`

### PR Checklist

- [ ] Wrote Tests
- [ ] Wrote & Generated Documentation (readme/natspec/dodoc)
- [ ] Ran `npm run lint` && `npm run lint:solidity` (solhint)
- [ ] Ran `npm run format` (prettier)
- [ ] Ran `npm run build`
- [ ] Ran `npm run test`

### CODE REVIEW

**Comment by Hugoo** [comment]
Task linked: [DEV-9554 Create CI to deploy contracts on release on production mainnet](https://app.clickup.com/t/2645698/DEV-9554)

**Review by b00ste — APPROVED** [review]
Nice hardhat task 😎

File: .github/workflows/deploy-verify.yml [hunk 1]
```diff
@@ -12,83 +14,45 @@ on:
...
+      - name: Build contract artifacts
+        run: npm run build
 
-      - name: Deploy + Verify ${{ matrix.contracts }} on Testnet
-        run: sh ./scripts/ci/deploy-verify.sh -n luksoTestnet -c ${{ matrix.contracts }}
+      - name: Verify Deployer Balance
+        run: npx hardhat verify-balance --network ${{ matrix.network }}
+
+      # We do not deploy the standard contracts on mainnet for the following reasons:
+      # 1) standard contracts are expensie to deploy on mainnet
```
Comment (Hugoo) [inline .github/workflows/deploy-verify.yml:42]:
```suggestion
      # 1) standard contracts are expensive to deploy on mainnet
```

File: hardhat.config.ts [hunk 2]
```diff
@@ -49,6 +51,21 @@ function getTestnetChainConfig(): NetworkUserConfig {
...
+  const config: NetworkUserConfig = {
+    live: true,
+    url: 'https://rpc.lukso.gateway.fm',
+    chainId: 42,
+    saveDeployments: true,
+  };
+
+  if (process.env.CONTRACT_VERIFICATION_MAINNET_PK !== undefined) {
+    config['accounts'] = [process.env.CONTRACT_VERIFICATION_MAINNET_PK];
+  }
```
Comment (Hugoo) [inline hardhat.config.ts:64]: Maybe you could add a else:

else: throw error: Missing env var etc.

(not a big deal, it just crossed my mind)

**Review by Hugoo — APPROVED** [review]
beautiful! super nice one ;)

File: hardhat.config.ts (hunk 2 above)
Comment (CJ42) [reply hardhat.config.ts:64]: Cannot be added, as other the build-lint-test CI job will fail (loading the hardhat config and cannot find this env variable)

**Pull request merged**